src/procedures/procedure.c

```diff
@@ -10,12 +10,17 @@
 #include "../RG.h"
 #include "../util/arr.h"
 #include "../util/rmalloc.h"
+#include "../util/strutil.h"
 #include "../graph/graphcontext.h"
 
 static rax *__procedures = NULL;
 
 static void _procRegister(const char *procedure, ProcGenerator gen) {
-	raxInsert(__procedures, (unsigned char *)procedure, strlen(procedure), gen, NULL);
+	char lower_proc_name[128];
+	size_t lower_proc_name_len = 128;
+	str_tolower(procedure, &lower_proc_name[0], &lower_proc_name_len);
```

Comment:
I think `&lower_proc_name[0]` can be replaced with just `lower_proc_name`
unless we get a compiler warning / error complaining that `char lower_proc_name[128]` isn't `char*`